### PR TITLE
Bug 1397934 - Fix ios10 deprecation warnings

### PR DESCRIPTION
**Pull request by q2r5**
Also hooked up local notifications to use actions (possibly implementing Bug 1364420)

### CODE REVIEW

**Comment by farhanpatel** [comment]
hey @q2r5 Sorry for the delay with this PR. The `UIColor` changes look good to me. But I think @jhugman would be the best person to double check the changes to UILocalNotification. I've pinged him to have a look 😄

File: Client/Application/AppDelegate.swift [hunk 1]
```diff
@@ -851,30 +853,29 @@ class AppSyncDelegate: SyncDelegate {
...
+                notificationContent.title = title
+                notificationContent.body = url.absoluteDisplayString
+                notificationContent.userInfo = [TabSendURLKey: url.absoluteString, TabSendTitleKey: title]
+                notificationContent.categoryIdentifier = TabSendCategory
+
+                let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
 
-                // Restore this when we fix Bug 1364420.
-                // notification.category = TabSendCategory
+                let request = UNNotificationRequest(identifer: TabSendCategory, content: notificationContent, trigger: trigger)
```
Comment (jhugman) [inline Client/Application/AppDelegate.swift]: Shouldn't this be `identifier`?

File: Client/Application/AppDelegate.swift [hunk 2]
```diff
@@ -851,30 +853,29 @@ class AppSyncDelegate: SyncDelegate {
...
-                notification.alertAction = nil
+                notificationContent.title = title
+                notificationContent.body = url.absoluteDisplayString
+                notificationContent.userInfo = [TabSendURLKey: url.absoluteString, TabSendTitleKey: title]
+                notificationContent.categoryIdentifier = TabSendCategory
+
+                let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
 
-                // Restore this when we fix Bug 1364420.
-                // notification.category = TabSendCategory
```
Comment (jhugman) [inline Client/Application/AppDelegate.swift]: [Bug 1364420][1] is about adding categories to push notifications from sent tabs: sent tabs come via sync commands, and the push notification is essentially a sync tickle which catches the send tab sync command. 

This change doesn't seem to add actions to the push notifications (see `NotificationService` and `FxAPushMessageHandler`), however, you may have identified a route to it. 

[1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1364420

File: Client/Application/AppDelegate.swift [hunk 3]
```diff
@@ -851,30 +853,29 @@ class AppSyncDelegate: SyncDelegate {
 
         // check to see what the current notification settings are and only try and send a notification if
         // the user has agreed to them
-        if let currentSettings = app.currentUserNotificationSettings {
-            if currentSettings.types.rawValue & UIUserNotificationType.alert.rawValue != 0 {
+        UNUserNotificationCenter.current.getNotificationSetings { settings in
+            if settings.alertSetting == UNNotificationSetting.enabled {
```
Comment (jhugman) [inline Client/Application/AppDelegate.swift]: Nit: prefer just `.enabled` to `UNNotificationSetting.enabled`

File: Client/Application/AppDelegate.swift [hunk 4]
```diff
@@ -851,30 +853,29 @@ class AppSyncDelegate: SyncDelegate {
...
-                notification.alertTitle = title
-                notification.alertBody = url.absoluteDisplayString
-                notification.userInfo = [TabSendURLKey: url.absoluteString, TabSendTitleKey: title]
-                notification.alertAction = nil
+                notificationContent.title = title
+                notificationContent.body = url.absoluteDisplayString
+                notificationContent.userInfo = [TabSendURLKey: url.absoluteString, TabSendTitleKey: title]
+                notificationContent.categoryIdentifier = TabSendCategory
+
+                let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
```
Comment (jhugman) [inline Client/Application/AppDelegate.swift]: Is this 1 second?

1. please use `.seconds(1)`
2. can we post the notification immediately? If the answer is no, please comment why.

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift [hunk 5]
```diff
@@ -145,13 +145,14 @@ extension BrowserViewController: WKNavigationDelegate {
         // prompting. On iOS9, depending on the scheme, iOS will prompt: "Firefox" wants to open "Twitter". It will ask
         // every time. There is no way around this prompt. (TODO Confirm this is true by adding them to the Info.plist)
 
-        let openedURL = UIApplication.shared.openURL(url)
-        if !openedURL {
-            let alert = UIAlertController(title: Strings.UnableToOpenURLErrorTitle, message: Strings.UnableToOpenURLError, preferredStyle: .alert)
-            alert.addAction(UIAlertAction(title: UIConstants.OKString, style: UIAlertActionStyle.default, handler: nil))
-            self.present(alert, animated: true, completion: nil)
+        UIApplication.shared.open(url, options: [:]) { success in
```
Comment (jhugman) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: Please could you open a followup bug to investigate possible use of `UIApplicationOpenURLOptionUniversalLinksOnly`

**Comment by q2r5** [comment]
Local notifications were tested using `AppSyncDelegate.displaySentTab`, bypassing the foreground check. The resulting notification had actions on force touch.

**Comment by farhanpatel** [comment]
@q2r5 sorry for the delay but James is on PTO. He will be back next week. Sorry for the delay

**Comment by jhugman** [comment]
This is good stuff! Thank you for your help. 

Many apologies for being on the PTO delay. I'd love to land this for 10.0 or 10.1!

Please could you rebase this against the current `HEAD` of master, then `needinfo` me on bugzilla?

Once you do that I can land it.

**Pull request closed without merging**